sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -639,7 +643,7 @@ object ScalaReflection extends ScalaReflection {
   def getConstructorParameters(cls: Class[_]): Seq[(String, Type)] = {
     val m = runtimeMirror(cls.getClassLoader)
     val classSymbol = m.staticClass(cls.getName)
-    val t = classSymbol.selfType
+    val t = selfType(classSymbol)
```

Comment:
Is this limited only to `selfType` ?

Reply:
Unfortunately, I do not have the technical knowhow to answer this question. What I can say is that `ScalaReflection` also uses `ClassSymbol.companion` which seems to work fine in this case. `ClassSymbol.annotations` fails, as might be expected, but it is not used anywhere in this class.